src/modules/roc_netio/target_uv/roc_netio/transceiver.cpp

```diff
@@ -51,92 +80,50 @@ Transceiver::Transceiver(packet::PacketPool& packet_pool,
     task_sem_.data = this;
     task_sem_initialized_ = true;
 
-    valid_ = true;
+    started_ = Thread::start();
 }
 
 Transceiver::~Transceiver() {
-    if (joinable()) {
-        roc_panic("transceiver: thread is not joined before calling destructor");
+    if (stop_sem_initialized_) {
+        if (int err = uv_async_send(&stop_sem_)) {
+            roc_panic("transceiver: uv_async_send(): [%s] %s", uv_err_name(err),
+                      uv_strerror(err));
+        }
+    } else if (task_sem_initialized_) {
```

Comment:
Let's replace `else if (task_sem_initialized_) {` with just `else {`:

* close_sems_() already has all necessary checks so we don't need to duplicate them;
* if more handles / semaphores will be added, the code will continue to work.

Reply:
Also, I think we should replace `if (stop_sem_initialized_) {` with `if (started_) {`. We should call uv_async_send() only if the thread was started; otherwise, we should close the semaphores in place.

Reply:
Done